rubocop/rubocop_rails.yml

```diff
@@ -3,6 +3,8 @@ inherit_from: rubocop.yml
 AllCops:
   Exclude:
     - db/schema.rb
+    - 'bin/**/*'
```

Comment:
I guess my one qualm with this is that there's no reason you couldn't/wouldn't have your own scripts in `bin/`, and presumably you'd want them to adhere to your style conventions as well. I can't conceive of a nice middle-ground through. 

Being explicit seems like a game you can't win, because even though there are the bin scripts from rails by default (which I guess would be a start) there can be any number of other scripts from other dependencies you may pull in.

Any thoughts on that?

Reply:
I was thinking along those lines as well. The problem is that there are multiple conflicting goals and there's no one solution (I can think of) that works for all of them. I've enumerated the goals I can think of below:

1. Files in `bin/` provided by Rails or other libraries shouldn't have to be restructured. Things like changing double quotes to single quotes are fine, but cops like `MethodLength` or `BlockLength` that require refactoring shouldn't apply.
2. Files in `bin/` may use different conventions than other files. For example, the default `update` script has the line `include FileUtils` which fails the `Style/MixinUsage` cop.
3. Users may write custom scripts and put them in `bin/`. These scripts should be linted (but see above about conventions).
4. When a user creates a new Rails project it shouldn't require major changes to pass Rubocop; running `rubocop --auto-correct` should fix any failing cops.
5. Simpler configurations of `rubocop.yml` are better than complicated ones.

Excluding `bin/` (which satisfies 1, 2, 4, and 5) seemed like the best solution to me when I was writing this PR since I was creating a new project and mostly concerned with 4 and 5.

Some other solutions I can think of are enumerating the files generated by Rails instead of excluding everything in `bin/` (1, 3, and 4) or excluding `bin/` only from `Style/MixinUsage` (2, 3, 4 (as of right now), and 5).

I think any of those solutions would work, but they do all involve some kind of tradeoff.

Reply:
Life is full of tradeoffs; no arguing with that. And as I said earlier, I don't think there's a nice/easy/maintainable middle-ground here, so I appreciate your thoughtful and thorough analysis of the tradeoffs and options here.

This works for me. Is there anything I'm missing that anyone else might want to chime in with? I'll leave this open for a few, just to make sure we capture and other feedback.